### PR TITLE
Document dynamic commands and bump version

### PR DESCRIPTION
**Pull request by sadmac7000**

### CODE REVIEW

File: argh/src/lib.rs [hunk 1]
```diff
@@ -166,6 +166,95 @@
...
+//!
+//! impl DynamicSubCommand for Dynamic {
+//!     fn commands() -> &'static [&'static CommandInfo] {
+//!         static RET: OnceCell<Vec<&'static CommandInfo>> = OnceCell::new();
+//!         RET.get_or_init(|| {
+//!             let mut commands = Vec::new();
+//!
+//!             // Do whatever dynamic process you want to populate your commands list here.
+//!             commands.push(&*Box::leak(Box::new(CommandInfo {
+//!                 name: "dynamic_command",
```
Comment (richkadel) [inline argh/src/lib.rs:219]: I think we either need to simplify this example a little bit, or explain it a little more. It looks like even your test cases aren't using `Box::leak()`. (Perhaps they should, so someone can, if they want to, copy you're test code as a starting point for their own implementation--assuming most use cases really need to leak a dynamic subcommand list in order to support the static lifetime references.) I'm assuming there isn't a _simpler_ way, and my concern is that most developers rarely if ever need to use Box::leak(). I think I understand why, but I think it would be helpful, for less advanced Rust programmers, to explain what you're doing.

The triple static lifetime referencing is confusing in its own right. Again, probably necessary to show a useful use case, but it's hard to follow.

Anything you can do to make this more clear would be great.

**Review by richkadel — CHANGES_REQUESTED** [review]
Some nits regarding my preference for 80 character columns, but the main thing is, I think the trait functions and/or the example code, or both, need additional explanation, because it's hard to follow the advanced (and uncommonly used) constructs of multiple static lifetimes, and Box::leak(). I can take another look this weekend if you want to follow up, or next week.

File: argh/src/lib.rs (hunk 1 above)
Comment (sadmac7000) [reply argh/src/lib.rs:226]: Made the comment a bit longer to explain some of these things.

**Review by richkadel — APPROVED** [review]
Looks good. Thanks!
